via/views/exceptions.py

```diff
@@ -61,16 +61,44 @@ def _get_meta(exception):
         if isinstance(exception, exception_type):
             return meta
 
-    return EXCEPTION_MAP.get(UnhandledException)
+    return EXCEPTION_MAP.get(UnhandledUpstreamException)
+
+
+@exception_view_config(
+    Exception,
+    route_name="proxy_google_drive_file",
+    renderer="via:templates/exception.html.jinja2",
+)
+@exception_view_config(
+    Exception,
+    route_name="proxy_google_drive_file:resource_key",
+    renderer="via:templates/exception.html.jinja2",
+)
+def google_drive_exceptions(exc, request):
+    """Catch all errors for Google Drive and display an HTML page."""
+
+    # Don't log 404's as they aren't very interesting
+    if not isinstance(exc, HTTPNotFound):
+        h_pyramid_sentry.report_exception()
+
+    return _get_error_body(exc, request)
 
 
 @exception_view_config(Exception, renderer="via:templates/exception.html.jinja2")
-@exception_view_config(HTTPError, renderer="via:templates/exception.html.jinja2")
 def all_exceptions(exc, request):
     """Catch all errors (Pyramid or Python) and display an HTML page."""
 
-    h_pyramid_sentry.report_exception()
+    # We don't want to log errors from upstream services or things which are
+    # the user goofing about making bad queries.
+    if not isinstance(
+        exc, (UpstreamServiceError, UnhandledUpstreamException, BadURL, HTTPClientError)
```

Comment:
I think it would be a good idea for these four to share a common base class like `ExternalRequestError` or something. Then you wouldn't need to list them all here, and if a new type of exception gets added you won't need to remember to add it here (assuming it is a subclass).

Reply:
`other_exceptions()` may be a more accurate name for `all_exceptions()` now

Reply:
It's a little awkward here. The two upstream ones definitely could. `BadURL` is sometimes raised as part of an upstream thing and sometimes locally. `HTTPClientError` just isn't ours at all.

I don't think it's too bad to just list the things we want here. It has the virtue of being dumb and clear for now.

Reply:
Renamed to `other_exceptions`